app/services/ak_user_params.rb

```diff
@@ -0,0 +1,33 @@
+require 'browser'
+
+class AkUserParams
+  def self.create(params)
+  # I'm assuming that the form params will come in with names that correspond to the AK API:
+    # akid
+    # email
+    # name
+      # name gets split to prefix, first name, middle name, last name and suffix automagically by AK ...
+    # address1
+    # address2
+    # city
+    # state
+    # zip
+    # postal
+    # country
+    # region
+    # phone
+    # mailing_id
+    # id
+    # plus4
+    # lang - get from current page session
+    # source - fwd, fb, tw, pr, mtl, taf
+    @user_params = params[:signature]
+    @user_params[:user_agent] = browser.user_agent
+    @user_params[:browser_detected] = browser.known?
+    @user_params[:mobile] = browser.mobile?
+    @user_params[:tablet] = browser.tablet?
+    @user_params[:platform] = browser.platform
+
+    @user_params
+  end
+end
```

Comment:
@Tuuleh `browser` is provided as a helper method, so is only available within a view/controller context. Can you inject the browser instance into `AKUserParams`.

Don't we want to persist an action directly in Champaign? We'd need an `action` model, where user agent info could be store. Then, you could simply pass the `action.id` along in message queue. The receiver could then get all the info it needs by accessing the action from the DB. What do you think?